ndcube/mixins/sequence_plotting.py

```diff
@@ -748,26 +727,29 @@ def __init__(self, seq, wcs=None, axes=None, plot_axis_indices=None,
             data_stack = np.stack([(cube.data * sequence_units[i]).to(data_unit).value
                                    for i, cube in enumerate(seq.data)])
         self.cumul_cube_lengths = np.cumsum(np.ones(len(seq.data)))
-        # Add dimensions of length 1 of concatenated data array
-        # shape for an missing axes.
-        if seq[0].wcs.naxis != len(seq.dimensions) - 1:
-            new_shape = list(data_stack.shape)
-            for i in np.arange(seq[0].wcs.naxis)[seq[0].missing_axes[::-1]]:
-                new_shape.insert(i + 1, 1)
-                # Also insert dummy coordinates and units.
-                axes_coordinates.insert(i + 1, None)
-                axes_units.insert(i + 1, None)
-            data_stack = data_stack.reshape(new_shape)
         # Add dummy axis to WCS object to represent sequence axis.
         new_wcs = utils.wcs.append_sequence_axis_to_wcs(wcs)
+        # Construct slices input.
+        if plot_axis_indices is None:
+            plot_axis_indices = [-1, -2]
+        # Currently, ArrayAnimator interprets slices in WCS order.
+        # So invert plot_axis_indices.
+        n_seq_dims = len(seq.dimensions)
+        slices_indices = utils.wcs.convert_between_array_and_pixel_axes(
+                np.array(plot_axis_indices), n_seq_dims)
+        slices = [0] * n_seq_dims
+        slices[slices_indices[-1]] = 'y'
+        slices[plot_axis_indices[0]] = 'x'
+        # Construct coord params input
+        if axes_units is not None:
+            coord_params = None  # In future construct axis unit helpers.
```

Comment:
@Cadair, I tried taking the ```axes_units``` input and construct ```CoordinateHelper``` objects to control the axes' unit, but couldn't make it work.  The online documentation I found didn't help a lot.  Any tips on how to do this?